From 280fd217a08eac937172376ba8b4daf98c10ddfe Mon Sep 17 00:00:00 2001
From: Dan Albert <danalbert@google.com>
Date: Thu, 20 Nov 2014 10:47:55 -0800
Subject: [PATCH 2/3] Add a pre-C++11 constexpr compatibility macro.

Needed for cases where something should be constexpr if possible, but
not being constexpr is fine if in pre-C++11 code (such as a const
static float member variable).

Bug: 18466763
Change-Id: I635d062575ba2fbc4cbe3a89f730128c404d95e1
Signed-off-by: arter97 <qkrwngud825@gmail.com>
---
 include/utils/Compat.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/utils/Compat.h b/include/utils/Compat.h
index eb8fa46..3d2e2a7 100644
--- a/include/utils/Compat.h
+++ b/include/utils/Compat.h
@@ -59,6 +59,17 @@ static inline ssize_t pread64(int fd, void* buf, size_t nbytes, off64_t offset)
 #endif
 
 /*
+ * Needed for cases where something should be constexpr if possible, but not
+ * being constexpr is fine if in pre-C++11 code (such as a const static float
+ * member variable).
+ */
+#if __cplusplus >= 201103L
+#define CONSTEXPR constexpr
+#else
+#define CONSTEXPR
+#endif
+
+/*
  * TEMP_FAILURE_RETRY is defined by some, but not all, versions of
  * <unistd.h>. (Alas, it is not as standard as we'd hoped!) So, if it's
  * not already defined, then define it here.
-- 
2.5.0

